From 8179be21e664cedb2e9d238cc2f6d04965e97275 Mon Sep 17 00:00:00 2001
From: Sergey Poznyakoff <gray@gnu.org>
Date: Thu, 11 May 2023 10:18:44 +0300
Subject: [PATCH] configure: Include needed header for major/minor macros

This helps in avoiding the warning about implicit function declaration
which is elevated as error with newer compilers e.g. clang 16

Signed-off-by: Khem Raj <raj.khem@gmail.com>

Upstream-Status: Backport
Signed-off-by: Ross Burton <ross.burton@arm.com>
---
 configure.ac | 18 ++++++++++++++++--
 1 file changed, 16 insertions(+), 2 deletions(-)

diff --git a/configure.ac b/configure.ac
index de479e7..c601029 100644
--- a/configure.ac
+++ b/configure.ac
@@ -43,8 +43,22 @@ AC_TYPE_UID_T
 AC_CHECK_TYPE(gid_t, int)
 
 AC_HEADER_DIRENT
-AX_COMPILE_CHECK_RETTYPE([major], [0])
-AX_COMPILE_CHECK_RETTYPE([minor], [0])
+AX_COMPILE_CHECK_RETTYPE([major], [0], [
+#include <sys/types.h>
+#ifdef MAJOR_IN_MKDEV
+# include <sys/mkdev.h>
+#endif
+#ifdef MAJOR_IN_SYSMACROS
+# include <sys/sysmacros.h>
+#endif])
+AX_COMPILE_CHECK_RETTYPE([minor], [0], [
+#include <sys/types.h>
+#ifdef MAJOR_IN_MKDEV
+# include <sys/mkdev.h>
+#endif
+#ifdef MAJOR_IN_SYSMACROS
+# include <sys/sysmacros.h>
+#endif])
 
 AC_CHECK_FUNCS([fchmod fchown])
 # This is needed for mingw build
-- 
2.34.1

